### ISSUE TITLE
Visualization Check of Datasets?

### ISSUE THREAD

**Issue by RylanSchaeffer** [MadryLab/constructed-datasets#7]
I'm visualizing the images in the tarball's d_robust folder and some of the images look odd, even by CIFAR10 standards:

![image](https://user-images.githubusercontent.com/8942987/72825498-233f9180-3c45-11ea-825e-0b719c0853dd.png)

![image](https://user-images.githubusercontent.com/8942987/72825509-276baf00-3c45-11ea-9618-6fb7f9a87e76.png)

Is this what the images look like when you visualize them? What's going on?

**Comment by RylanSchaeffer** [comment]
Like, what the hell are these supposed to be?

![image](https://user-images.githubusercontent.com/8942987/72825556-40746000-3c45-11ea-8c8d-0d1e1704da08.png)

![image](https://user-images.githubusercontent.com/8942987/72825567-4407e700-3c45-11ea-8c54-6cff78034cf8.png)

![image](https://user-images.githubusercontent.com/8942987/72825575-49653180-3c45-11ea-95b4-eb345769d769.png)

**Comment by andrewilyas** [comment]
Hi, these images are generated by the inversion procedure described in this paper: https://arxiv.org/abs/1906.00945, run on the training images. The results get more realistic as the number of steps are increased and aren't perfect---here we just used the minimum number of steps for reasonable accuracy.

**Comment by RylanSchaeffer** [comment]
@andrewilyas , I'm asking about https://arxiv.org/abs/1905.02175, not https://arxiv.org/abs/1906.00945

The non-robust example offered in the "Adversarial Examples Are Not Bugs, They Are Features" paper:

![image](https://user-images.githubusercontent.com/8942987/72924505-0aa3aa00-3d1f-11ea-9e7a-ff853b39b08d.png)

This is what the non-robust images look like when I load them:

![image](https://user-images.githubusercontent.com/8942987/72924441-ecd64500-3d1e-11ea-81cc-f465b1f01d21.png)

Am I making a mistake? Or is this how the non-robust images appear as to you too?

**Comment by RylanSchaeffer** [comment]
The readme says that the tarball contains data used in https://arxiv.org/abs/1905.02175. Is this not correct?

**Comment by RylanSchaeffer** [comment]
Here's a minimum working example for how I'm generating these images

```
import numpy as np
import os
import torch


# set to your nonrobust data path
nonrobust_dataset_path = 'release_datasets/d_non_robust_CIFAR'

# load data per README instructions
nonrobust_images = torch.cat(torch.load(os.path.join(nonrobust_dataset_path,
                                                 f"CIFAR_ims")))

# convert to numpy
nonrobust_images = nonrobust_images.numpy()


# transpose images from (num samples, 3, height, width) to (num samples, height, width, 3)
# this is for matplotlib
nonrobust_images = np.transpose(nonrobust_images, (0, 2, 3, 1))


# specify how many examples to plot
num_rows, num_cols = 3, 5
fig, axs = plt.subplots(num_rows, num_cols)


# choose random images
random_image_indices = np.random.choice(
    a=np.arange(nonrobust_images.shape[0]),
    size=num_rows * num_cols,
    replace=False)


for i, idx in enumerate(random_image_indices):
    ax = axs[i // 5, i % 5]
    image = nonrobust_images[idx]
    ax.imshow(image)

    # hide tick marks
    ax.set_yticklabels([])
    ax.set_xticklabels([])

plt.show()
```

**Comment by RylanSchaeffer** [comment]
example output: 
![image](https://user-images.githubusercontent.com/8942987/72927334-ff06b200-3d23-11ea-8ba9-f8078d0aeaba.png)

![image](https://user-images.githubusercontent.com/8942987/72927360-07f78380-3d24-11ea-83f8-2a0969b10419.png)

![image](https://user-images.githubusercontent.com/8942987/72927370-0ded6480-3d24-11ea-936d-c059b47f254d.png)

**Comment by erhul** [comment]
This should be correct, and this is what I visualized.

<img width="1530" height="459" alt="Image" src="https://github.com/user-attachments/assets/be92a492-7b2c-4d6d-a03f-02cc9a510c16" />
